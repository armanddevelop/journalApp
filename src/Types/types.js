export const types = {
  login: "[Auth] Login",
  logout: "[Auth] LogOut",
  uiSetError: "[UI] setError",
  uiRemoveError: "[UI] removeError",
  uiStartLoading: "[UI] Start loading",
  uiFinishLoading: "[UI] Finish loading",
  uiErrorInFirebase: "[UI] Error logIn",
  uiRemoveErrorInFireBase: "[UI] Remove Error logIn",
};
