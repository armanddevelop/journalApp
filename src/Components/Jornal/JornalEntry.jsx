import moment from "moment";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import { notesActiveAction } from "../../Actions/notes";

export const JornalEntry = ({ title, body, date, url, id }) => {
  const noteDate = moment(date);
  const dispatch = useDispatch();
  const handleEntryClick = () => {
    const noteObj = { title, body, date, url };
    dispatch(notesActiveAction(id, noteObj));
  };
  return (
    <div className="jornal__entry pointer" onClick={handleEntryClick}>
      {url && (
        <div
          className="jornal__entry-picture"
          style={{
            backgroundSize: "cover",
            backgroundImage: `url(${url})`,
          }}
        ></div>
      )}
      <div className="jornal__entry-body">
        <p className="jornal__entry-title">{title}</p>
        <p className="jornal__entry-content">{body}</p>
      </div>
      <div className="jornal__entry-date-box">
        <span>{noteDate.format("dddd")}</span>
        <h4>{noteDate.format("Do")}</h4>
      </div>
    </div>
  );
};

JornalEntry.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string,
};
