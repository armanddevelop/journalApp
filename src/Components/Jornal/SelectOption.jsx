export const SelectOption = () => {
  return (
    <div className="nothing__main-content">
      <p>
        Select an Item
        <br />
        or Create a new Entry
      </p>
      <i className="far fa-star fa-4x mt-5"></i>
    </div>
  );
};
