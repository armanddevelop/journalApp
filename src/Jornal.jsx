import { AppRouter } from "../src/Routers/AppRouter";

export const Jornal = () => {
  return (
    <div>
      <AppRouter />
    </div>
  );
};
