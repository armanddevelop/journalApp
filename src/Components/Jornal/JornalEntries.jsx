import { JornalEntry } from "./JornalEntry";

export const JornalEntries = () => {
  const entries = [1, 2, 3, 4, 5];
  return (
    <div className="jornal__entries">
      {entries.map((entry) => (
        <JornalEntry key={entry} />
      ))}
    </div>
  );
};
