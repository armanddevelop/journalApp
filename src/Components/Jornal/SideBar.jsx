import { useDispatch, useSelector } from "react-redux";
import { startLogOut } from "../../Actions/auth";
import { startNewNoteAction } from "../../Actions/notes";
import { JornalEntries } from "./JornalEntries";

export const SideBar = () => {
  const dispatch = useDispatch();
  const { name } = useSelector((state) => state.auth);
  const handleLogOut = () => {
    dispatch(startLogOut());
  };
  const handleNewEntry = () => {
    console.log("licha");
    dispatch(startNewNoteAction());
  };
  return (
    <aside className="jornal__sidebar">
      <div className="jornal__sidebar-navbar">
        <h3 className="mt-5">
          <i className="far fa-moon"></i>
          <span> {name}</span>
        </h3>
        <button className="btn" onClick={handleLogOut}>
          LogOut
        </button>
      </div>
      <div className="jornal__new-entry" onClick={handleNewEntry}>
        <i className="far fa-calendar-plus fa-5x jornal__new-entry-icon"></i>
        <p className="mt-5">New Entry</p>
      </div>
      <JornalEntries />
    </aside>
  );
};
