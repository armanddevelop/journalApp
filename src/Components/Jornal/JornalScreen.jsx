import { SelectOption } from "./SelectOption";
import { useSelector } from "react-redux";
import { NoteScreen } from "../Notes/NoteScreen";
import { SideBar } from "./SideBar";
export const JornalScreen = () => {
  const { active } = useSelector((state) => state.notes);
  return (
    <div className="jornal__main-content">
      <SideBar />
      <main>{!active ? <SelectOption /> : <NoteScreen />}</main>
    </div>
  );
};
