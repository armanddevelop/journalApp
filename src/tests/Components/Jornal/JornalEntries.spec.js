import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { JornalEntries } from "../../../Components/Jornal/JornalEntries";
import { types } from "../../../Types/types";

const initialState = {
  notes: {
    notes: [
      {
        title: "una nota",
        body: "Esta es una nota de prueba",
        id: "123456",
        date: 87926,
      },
      {
        title: "NOTA",
        body: "Otra nota de prueba",
        id: "123",
        date: 65412,
      },
    ],
  },
};
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore(initialState);

describe("Test in JornalEntries component", () => {
  const wrapper = mount(
    <Provider store={store}>
      <JornalEntries />
    </Provider>
  );
  it("SnapShot to JornalEntries component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should no exist jornal__entry-picture className", () => {
    const entryPicture = wrapper.find(".jornal__entry-picture").exists();

    expect(entryPicture).toBe(false);
  });

  it("Should execute handleEntryClick", () => {
    wrapper.find(".jornal__entry.pointer").first().prop("onClick")();
    const actions = store.getActions();
    const expectedObject = {
      id: expect.any(String),
      title: expect.any(String),
      body: expect.any(String),
      date: expect.any(Number),
      url: undefined,
    };

    expect(actions[0].type).toEqual(types.notesActive);
    expect(actions[0].payload).toEqual(expectedObject);
  });

  it("Should display the contet of note in component", () => {
    const title = wrapper.find(".jornal__entry-title").first().text();
    const body = wrapper.find(".jornal__entry-content").first().text();
    const date = wrapper.find(".jornal__entry-date-box").first().text();

    expect(title).toEqual(initialState.notes.notes[0].title);
    expect(body).toEqual(initialState.notes.notes[0].body);
    expect(date).toEqual("Wednesday31st");
  });
});
