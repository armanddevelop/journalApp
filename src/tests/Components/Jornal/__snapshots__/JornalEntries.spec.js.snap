// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in JornalEntries component SnapShot to JornalEntries component 1`] = `
<div
  className="jornal__entries"
>
  <div
    className="jornal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="jornal__entry-body"
    >
      <p
        className="jornal__entry-title"
      >
        una nota
      </p>
      <p
        className="jornal__entry-content"
      >
        Esta es una nota de prueba
      </p>
    </div>
    <div
      className="jornal__entry-date-box"
    >
      <span>
        Wednesday
      </span>
      <h4>
        31st
      </h4>
    </div>
  </div>
  <div
    className="jornal__entry pointer animate__animated animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="jornal__entry-body"
    >
      <p
        className="jornal__entry-title"
      >
        NOTA
      </p>
      <p
        className="jornal__entry-content"
      >
        Otra nota de prueba
      </p>
    </div>
    <div
      className="jornal__entry-date-box"
    >
      <span>
        Wednesday
      </span>
      <h4>
        31st
      </h4>
    </div>
  </div>
</div>
`;
